src/test/java/com/example/onlinebookstore/service/book/BookServiceImplTest.java

```diff
@@ -0,0 +1,62 @@
+package com.example.onlinebookstore.service.book;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.example.onlinebookstore.dto.book.BookResponseDto;
+import com.example.onlinebookstore.mapper.BookMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.repository.BookRepository;
+import java.math.BigDecimal;
+import java.util.List;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceImplTest {
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    @DisplayName("Get all books")
+    public void findAll_ValidPageable_ReturnsAllBooks() {
+        Book book = new Book();
+        book.setId(1L);
+        book.setTitle("Harry Potter");
+        book.setAuthor("J. K. Rowling");
+        book.setIsbn("9781111111111");
+        book.setPrice(BigDecimal.valueOf(100));
+
+        BookResponseDto bookResponseDto = new BookResponseDto();
+        bookResponseDto.setId(book.getId());
+        bookResponseDto.setTitle(book.getTitle());
+        bookResponseDto.setAuthor(book.getAuthor());
+        bookResponseDto.setIsbn(book.getIsbn());
+        bookResponseDto.setPrice(book.getPrice());
+
+        Pageable pageable = PageRequest.of(0,10);
+        List<Book> books = List.of(book);
+        Page<Book> bookPage = new PageImpl<>(books, pageable, books.size());
+
+        List<BookResponseDto> expected = List.of(bookResponseDto);
+
+        Mockito.when(bookRepository.getAll(pageable)).thenReturn(bookPage);
+        Mockito.when(bookMapper.toDto(book)).thenReturn(bookResponseDto);
+
+        List<BookResponseDto> actual = bookService.findAll(pageable);
+
+        assertEquals(expected, actual);
```

Comment:
what about [assertIterableEquals](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertIterableEquals-java.lang.Iterable-java.lang.Iterable-)([Iterable](https://docs.oracle.com/javase/8/docs/api/java/lang/Iterable.html?is-external=true)<?> expected, [Iterable](https://docs.oracle.com/javase/8/docs/api/java/lang/Iterable.html?is-external=true)<?> actual)